lib/insights_cloud/async/vmaas_reposcan_sync.rb

```diff
@@ -0,0 +1,63 @@
+require 'rest-client'
+
+module InsightsCloud
+  module Async
+    # Triggers VMaaS reposcan sync via IoP gateway when repositories are synced
+    class VmaasReposcanSync < ::Actions::EntryAction
+      include ::ForemanRhCloud::CertAuth
+
+      # Subscribe to Katello repository sync hook action, if available
+      def self.subscribe
+        'Actions::Katello::Repository::SyncHook'.constantize
+      rescue NameError
+        Rails.logger.debug('VMaaS reposcan sync: Repository::SyncHook action not found')
+        nil
+      end
+
+      def plan(repo, *_args)
+        repo_id = repo.is_a?(Hash) ? (repo[:id] || repo['id']) : nil
+        unless repo_id
+          logger.error("VMaaS reposcan sync: missing repository id in SyncHook plan parameters: #{repo.inspect}")
+          return
+        end
+        return unless ::ForemanRhCloud.with_iop_smart_proxy?
+
+        plan_self
+      end
+
+      def run
+        url = ::InsightsCloud.vmaas_reposcan_sync_url
+
+        response = execute_cloud_request(
+          method: :put,
+          url: url,
+          headers: { 'Content-Type' => 'application/json' }
+        )
+
+        if response.code >= 200 && response.code < 300
+          logger.info("VMaaS reposcan sync triggered successfully: #{response.code}")
+        else
+          logger.error("VMaaS reposcan sync failed with status: #{response.code}, body: #{response.body}")
+        end
+
+        response
+      rescue RestClient::ExceptionWithResponse => e
+        logger.error("VMaaS reposcan sync failed: #{e.response&.code} - #{e.response&.body}")
+        raise
+      rescue StandardError => e
+        logger.error("Error triggering VMaaS reposcan sync: #{e.message}, response: #{e.respond_to?(:response) ? e.response : nil}")
+        raise
```

Comment:
Instead of just logging, I think it would be useful to put these in the task output, e.g. `output[:message] = xxxx` This way it will be visible in the Dynflow console if we need to troubleshoot.

Reply:
I’ve added both logging and setting output[:message], so the messages appear in the Dynflow console and also in the logs. I can drop the logging if you prefer to avoid duplication.

Reply:
I think both are useful! let's leave them